### PR TITLE
Add docs how to create a custom values file

### PR DESCRIPTION
**Pull request by kota2and3kan**
As part of re-organizing the document of [scalar-kubernetes](https://github.com/scalar-labs/scalar-kubernetes), I added/updated the document of Scalar Helm Charts.

This PR adds new documents `Configure a custom values file for Scalar Helm Chart`.
These documents do not cover all parameters since there are some parameters that users don't need to configure.
These documents describe `Required configurations` and `Optional configurations` from the perspective of users.

* Configure a custom values file for Scalar Helm Chart (root document)
    * Configure a custom values file for Envoy
    * Configure a custom values file for ScalarDB Server
    * Configure a custom values file for ScalarDB GraphQL
    * Configure a custom values file for ScalarDL Ledger
    * Configure a custom values file for ScalarDL Auditor
    * Configure a custom values file for ScalarDL Schema Loader
    * Configure a custom values file for Scalar Manager
* How to use Secret resources to pass the credentials as the environment variables into the properties file (re-organize)
* Mount key/certificate files to the container in ScalarDL Helm Charts (re-organize)

Also, this PR separates/re-organizes the existing document `How to use a customized properties file with Scalar Helm Charts` since there are some duplicated contents.

* How to use a customized properties file with Scalar Helm Charts (Separated and Removed)
    * `Set your properties file to a custom values file` section -> Move to each new document (Configure a custom values file of XXXX)
    * `Use Secret resources to pass the credentials as the environment variables into the properties file` section -> Move to the new document (How to use Secret resources to pass the credentials as the environment variables into the properties file)
    * `Mount arbitrary files like key and certificate files to the container in Scalar DL Helm Charts` section -> Move to the new document (Mount key/certificate files to the container in ScalarDL Helm Charts)
    * `Set flags with the Scalar DL Schema Loader chart` -> Move to the new document (Configure a custom values file of ScalarDL Schema Loader)

Please take a look!

### CODE REVIEW

**Comment by kota2and3kan** [comment]
@supl  
These new documents describe the details of the parameters in each chart.
And, there are two sections `Required configurations` and `Optional configurations`.

Since I'm not familiar with the Scalar Manager, could you please take a look at Scalar Manager's document (`docs/configure-custom-values-scalar-manager.md` file) from the perspective of each configuration is `required` or `optional`?

**Review by supl — APPROVED** [review]
Thank you.
I checked the required and optional custom values in the document.
It looks good to me.

File: docs/configure-custom-values-scalardb.md [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+
+This document explains how to create your custom values file for the ScalarDB Server chart. If you want to know the details of the parameters, please refer to the [README](https://github.com/scalar-labs/helm-charts/blob/main/charts/scalardb/README.md) of the ScalarDB Server chart.
+
+## Required configurations
+
+### Scalar Envoy configurations
+
+You must set the Scalar Envoy configurations in the custom values file for ScalarDB Server.
+
+If you deploy ScalarDB Server on a Kubernetes environment, the client requests are sent to ScalarDB Server via Scalar Envoy as a load balancer for gRPC requests. In other words, the Scalar Envoy pods listen the client requests.
```
Comment (feeblefakie) [inline docs/configure-custom-values-scalardb.md:11]:
```suggestion
```

**Review by feeblefakie — COMMENTED** [review]
Thank you and sorry for the late review 🙇 
Took a look at the ScalarDB doc and left some comments.
PTAL when you get a chance.

File: docs/configure-custom-values-scalardb.md [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    limits:
+      cpu: 2000m
+      memory: 4Gi
+```
+
+### Secret configurations
+
+If you want to use environment variables to set some properties (e.g., credentials) in the `scalardb.databaseProperties`, you can use `scalardb.secretName` to specify the Secret resource that includes some credentials.
+
+For example, you can set credentials for a backend database (`scalar.db.username` and `scalar.db.password`) using environment variables. It makes your pods more secure.
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/configure-custom-values-scalardb.md]: Thank you for your suggestion!
I applied the same suggestion to other documents in the following commit.
https://github.com/scalar-labs/helm-charts/pull/166/commits/4604f47e187adb4ff471fa779271c87364e13a95

File: docs/configure-custom-values-scalardb.md [hunk 3]
```diff
@@ -0,0 +1,195 @@
+# Configure a custom values file for ScalarDB Server
+
+This document explains how to create your custom values file for the ScalarDB Server chart. If you want to know the details of the parameters, please refer to the [README](https://github.com/scalar-labs/helm-charts/blob/main/charts/scalardb/README.md) of the ScalarDB Server chart.
+
+## Required configurations
+
+### Scalar Envoy configurations
+
+You must set the Scalar Envoy configurations in the custom values file for ScalarDB Server.
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/configure-custom-values-scalardb.md]: Thank you for your suggestion!
I applied the same suggestion to other documents in the following commit.
https://github.com/scalar-labs/helm-charts/pull/166/commits/4ca070b3e74f28f23c579d341bd65b1f3f2ae63a

File: docs/configure-custom-values-scalardb.md [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+You can specify the number of replicas (pods) of ScalarDB Server using `scalardb.replicaCount`.
+
+```yaml
+scalardb:
+  replicaCount: 3
+```
+
+### Resource configurations
+
+If you want to control pod resources using the requests and limits of Kubernetes, you can use `scalardb.resources`.
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/configure-custom-values-scalardb.md:73]: Ah, I see. That's right. We should explain the limitation of pay-as-you-go containers.
I added the following descriptions in the commit https://github.com/scalar-labs/helm-charts/pull/166/commits/a60d9663c3f1fa0c90f67093e2c69666d33a5aae.

```
Note that the resources for one pod of Scalar products are limited to 2vCPU / 4GB memory from the perspective of the commercial license. Also, when you get the pay-as-you-go containers provided from AWS Marketplace, you cannot run those containers with more than 2vCPU / 4GB memory configuration in the `resources.limits`. When you exceed this limitation, pods are automatically stopped.
```

File: docs/configure-custom-values-scalardb.md [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+### Replica configurations
+
+You can specify the number of replicas (pods) of ScalarDB Server using `scalardb.replicaCount`.
+
+```yaml
+scalardb:
+  replicaCount: 3
+```
+
+### Resource configurations
```
[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/configure-custom-values-scalardb.md:71]: @feeblefakie  
I agree with your concerns. Indeed, I think they are unfriendly/misleading.

I think we can explain `updating is recommended`, `default value is recommended`, and `truly optional based on user's environment` as follows.

* Pattern A (Add notes in each section title)
  ```
  ## Required configurations
  ...

  ## Optional configurations

  ### Resource configurations (Recommended in the production environment)
  ### Secret configurations (Recommended in the production environment)
  ### Affinity configurations (Recommended in the production environment)
  ### Taints/Tolerations configurations (Recommended in the production environment)
  ### Prometheus/Grafana configurations (Recommended in the production environment)

  ### SecurityContext configurations (Default value is recommended)

  ### Replica configurations (Optional based on your environment)
  ### Logging configurations (Optional based on your environment)
  ```
* Pattern B (Separate `Optional configurations` section more finely)
  ```
  ## Required configurations
  ...

  ## Recommended configurations in the production environment
  ### Resource configurations
  ### Secret configurations
  ### Affinity configurations
  ### Taints/Tolerations configurations
  ### Prometheus/Grafana configurations
  
  ## Not recommended to update from default configurations (Default values are recommended)
  ### SecurityContext configurations
  
  ## Optional configurations based on your environment
  ### Replica configurations
  ### Logging configurations
  ```

In pattern B, users can know whether it is recommended or not per section.
And, pattern A is a bit duplicated.
So, I think the pattern B is better.

Which pattern do you think is better?
I will restructure the documents including other products after discussion.

**Review by kota2and3kan — COMMENTED** [review]
@feeblefakie  
Thank you for your review!
I applyed some suggestions and left some comments.

I want to discuss how to explain the `updating is recommended`, `default value is recommended`, and `truly optional based on user's environment` in this document.

Please take a look when you have time!

File: docs/configure-custom-values-scalardb.md (hunk 5 above)
[the remark this answers is not in the document]
Comment (feeblefakie) [reply docs/configure-custom-values-scalardb.md:71]: Both are good, but Pattern A looks slightly better to me! Thank you!

[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/configure-custom-values-scalardb.md:71]: Thank you for your comment! I will update these documents based on pattern A.

[the remark this answers is not in the document]
Comment (kota2and3kan) [reply docs/configure-custom-values-scalardb.md]: @feeblefakie  
I added the notes in each section title in all documents based on pattern A.
Please take a look again when you have time!

**Review by feeblefakie — APPROVED** [review]
LGTM! Thank you!
(Very sorry for the late review 🙇 )

**Pull request merged**